### ISSUE TITLE
Change default anchor presets and Container sizing to fill parent.

### ISSUE THREAD

**Issue by omid3098** [godotengine/godot-proposals#8454]
### Describe the project you are working on

An application that uses Godot as GUI 

### Describe the problem or limitation you are having in your project

I am not sure if there is a such thing as best practices to design a UI for applications but here is how I usually try to implement a UI structure like the way I do designing a flutter application:
- Create a parent to cover the whole area with full-rect anchor preset (even if the most parts remain empty) 
<img width="130" alt="image" src="https://github.com/godotengine/godot-proposals/assets/6388730/a893a558-3f3f-423f-aefe-e7dc302eaef5">

- Create regions for header and body with full-rect anchor preset or Container sizing to fill+expand horizontally and vertically
<img width="341" alt="image" src="https://github.com/godotengine/godot-proposals/assets/6388730/b49fd4da-5d2e-4679-af60-a7f260f1442a">

- Go deep down in the tree and add all nodes I want and apply these presets/parameters manually. 
For example the case I want to add a close button to the top right corner of my GUI:
Normal approach (faster for mockups or MVP)
- Create a contol node and set it to full rect
- Create a button and set the anchors to top-right
- Add a x lable under the button and set anchors to full rect

My approach: (Slower but scalable on expanding/complex GUI projects)
- Create a contol node and set it to full rect
- Create a VBox contanert and set the anchors to full rect
- Create a pannel under Vbox container for header and set Layout Container Sizing to fill/expand horizontally and vertically
- Set the header stretch ration to 0.1
- Create a pannel under Vbox container for body and set Layout Container Sizing to fill/expand horizontally and vertically (this will automatically fill the 0.9 of the parent)
- Add a HBox container under the header to put header contents
- Add the button and lable under that and align the header to left.

Pros: each one of these elements can be packed into a scene and be used in other scenes of the application.
Cons: Slower to set all anchors and sizings manually

Overal structure:
<img width="221" alt="image" src="https://github.com/godotengine/godot-proposals/assets/6388730/20a6455d-d6ee-4ec0-9d80-2fe3edde4907">

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Set the default anchor for UI elements to full rect and the default content sizing to fill/expand horizontally and vertically.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

If the default anchor is set to full rect and the default content sizing is set to fill/expand horizontally and vertically, we can easily design very complex UIs that are scalable and re-usable by packing them as separate scenes and the end result will be visible to the user much faster. (if they design with explained workflow like the way flutter does)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

a work around would be to assign hotkeys to set these things. and in the meantime I plan to implement a plugin to do so.

### Is there a reason why this should be core and not an add-on in the asset library?

I believe this UI design workflow suites better as default way to design UI in Godot.

**Comment by AThousandShips** [comment]
This will break compatibility unless it's done in a way that handles it correctly, as the default case isn't stored, AFAIK

**Comment by Calinou** [comment]
> This will break compatibility unless it's done in a way that handles it correctly, as the default case isn't stored, AFAIK

We can apply new defaults when creating new nodes in the editor only, but this creates an inconsistency between nodes created in the editor and nodes created using a script (including `@tool` scripts).

**Comment by YuriSizov** [comment]
I agree that the fill anchor preset is one that is used the most. However, as of Godot 4.0 the default layout mode is just position and size, which is equivalent to top left. This is meant to be the most trivial to use mode. So we can't really change the default anchor mode to fill, because it would completely remove the simplest layout option or make it harder to find than just anchors.

**Comment by omid3098** [comment]
> Either way, I'm not 100% convinced that the proposed default is a straight upgrade in all cases. Game UIs are often fairly simple (especially for mobile games), so designing around extensibility at the cost of ease of use is not always worth it.

The cost is not ease of use. currently the behaviour is "I put the anchors in the upper-left. change it however you want" and we almost always want to change it! my suggestion is "I put the anchors in full-rect. change it however you want".
I have also watched many tutorials to design UI in Godot and almost none of them are using the default anchors.
Also, if we put the casual games and game-jams aside, I mostly observed and developed complex UIs for mobile games.

> I agree that the fill anchor preset is one that is used the most. However, as of Godot 4.0 the default layout mode is just position and size, which is equivalent to top left. This is meant to be the most trivial to use mode. So, we can't really change the default anchor mode to fill, because it would completely remove the simplest layout option or make it harder to find than just anchors.

I totally understand there are reasons you had to do the way it currently is. I just thought this behaviour is more usable and needed to be told :)
